src/main/scala/com/github/jberggg/router/RouterDsl.scala

```diff
@@ -0,0 +1,33 @@
+package com.github.jberggg.router
+
+import cats.implicits._
+import cats.effect.Async
+import org.scalajs.dom.window
+import fs2.concurrent.Channel
+import fs2.Stream
+import org.http4s.Uri.Path
+
+import Domain._
+
+trait RouterDsl[F[_]] {
+
+    def requestedPaths: Stream[F, Path]
+
+    def navigate(to: Path): F[Unit]
+
+}
+
+object RouterDsl {
+    def interpreter[ F[_] : Async ](pathChannel: Channel[F, Path]): RouterDsl[F] = new RouterDsl[F] {
+           
+        override def requestedPaths: Stream[F,Path] = pathChannel.streamAndRegisterEventListener
+            
+        override def navigate(to: Path): F[Unit] =
+            Async[F].delay( window.location.hash = (to.renderString) ) *>
+            pathChannel.send(to).void
```

Comment:
So actually I don't know how the browser API works here. When you change the value of `window.location.hash`, does that cause a `HashChangeEvent` to fire? In which case I would think sending `to` to the `Channel` would actually cause it to be duplicated.

Reply:
I tested it through, although it seems like there are no duplicates or the fs2 constructs filter them out, setting the hash alone is sufficient. I removed the explicit call which pushes the new location-hash.

Reply:
Ah, interesting! Thanks for investigating that.